internal/polkavm/common.go

```diff
@@ -137,7 +137,7 @@ func (m *Memory) Sbrk(size uint64) (uint64, error) {
 }
 
 func (m *Memory) allocatePages(startPage uint64, count uint64) {
-	required := (startPage + count) * PageSize
+	required := ((startPage + count) * PageSize) - m.rw.address
```

Comment:
We were calculating required based on starting at address zero, so allocating more than needed.